msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs

```diff
@@ -156,6 +158,30 @@ public override bool Execute ()
 			return !Log.HasLoggedErrors;
 		}
 
+		void AddXamarinVersionNumber (PDictionary plist)
+		{
+			// Add our own version number
+			if (IsWatchApp)
+				return;
+
+			string value;
+
+			// This key is our supported way of determining if an app
+			// was built with Xamarin, so it needs to be present in all apps.
+			if (TargetFramework.IsDotNet) {
+				value = DotNetVersion;
+			} else if (Platform != ApplePlatform.MacOSX) {
+				var version = Sdks.XamIOS.ExtendedVersion;
+				value = string.Format ("{0} ({1}: {2})", version.Version, version.Branch, version.Hash);
+			} else {
+				return;
+			}
+
+			var dict = new PDictionary ();
+			dict.Add ("Version", new PString (value));
+			plist.Add ("com.xamarin.ios", dict);
```

Comment:
was it `*.ios` for all SDK (e.g. for tvOS) ? I never realized it

OTOH if this is now shared with XM (for legacy) and the version number is quite different then I think it's worth having a _correct_ key.

Reply:
Yes, it was always "com.xamarin.ios", even for watchOS and tvOS (but we never added this key for macOS).

I went back and forth whether to add a new key or not (or even have two keys: the old and a new one), and in the end decided to keep the old one (only), just to be compatible with existing code, even though it looks somewhat ugly - mostly because there won't be any compiler errors/warnings to help developers.

That said, I can certainly change the key, knowing that other people vote for an updated name :)

Reply:
I'm _fine_ with the key, mostly so I don't have to vote on a new one...

Reply:
> mostly because there won't be any compiler errors/warnings to help developers.

Was it ever documented ? it so it should be updated, if not then it's likely not used very much and you have a better chance to _promote_ the feature (in any form) in the breaking change document.

Since it would be new for XM/Legacy and the version number is different from XI then `com.xamarin.mac` might be better (as in less confusing).

For ,net you could use a single key like `com.microsoft.sdk` since the OS is not really hard to guess from other elements (inside the plist or at runtime).

TBH with the addition (for dotnet) of the version numbers inside the assembly name this is much less an issue than it was before - so removing it (for dotnet) would also be a possibility.

Reply:
I've changed it to use `com.microsoft.<platform>` now.